Mathlib/Analysis/SpecialFunctions/Trigonometric/Bounds.lean

```diff
@@ -53,6 +49,75 @@ theorem sin_lt {x : ℝ} (h : 0 < x) : sin x < x := by
   norm_num
 #align real.sin_lt Real.sin_lt
 
+lemma sin_le (hx : 0 ≤ x) : sin x ≤ x := by
+  obtain rfl | hx := hx.eq_or_lt
+  · simp
+  · exact (sin_lt hx).le
+
+lemma lt_sin (hx : x < 0) : x < sin x := by simpa using sin_lt <| neg_pos.2 hx
+lemma le_sin (hx : x ≤ 0) : x ≤ sin x := by simpa using sin_le <| neg_nonneg.2 hx
+
+lemma cos_quadratic_lower_bound : 1 - x ^ 2 / 2 ≤ cos x := by
+  wlog hx₀ : 0 ≤ x
+  · simpa using this $ neg_nonneg.2 $ le_of_not_le hx₀
+  suffices MonotoneOn (fun x ↦ cos x + x ^ 2 / 2) (Ici 0) by
+    simpa using this left_mem_Ici hx₀ hx₀
+  refine monotoneOn_of_hasDerivWithinAt_nonneg (convex_Ici _) (Continuous.continuousOn $ by
+    continuity) (fun x _ ↦
+    ((hasDerivAt_cos ..).add $ (hasDerivAt_pow ..).div_const _).hasDerivWithinAt) fun x hx ↦ ?_
+  simpa [mul_div_cancel_left] using sin_le $ interior_subset hx
+
+/-- **Jordan's inequality**. -/
+lemma two_div_pi_mul_le_sin (hx₀ : 0 ≤ x) (hx : x ≤ π / 2) : 2 / π * x ≤ sin x := by
+  rw [← sub_nonneg]
+  suffices : ConcaveOn ℝ (Icc 0 (π / 2)) (fun x ↦ sin x - 2 / π * x)
+  · refine (le_min ?_ ?_).trans $ this.min_le_of_mem_Icc ⟨hx₀, hx⟩ <;> field_simp
+  exact concaveOn_of_hasDerivWithinAt2_nonpos (convex_Icc ..)
+    (Continuous.continuousOn $ by continuity)
+    (fun x _ ↦ ((hasDerivAt_sin ..).sub $ (hasDerivAt_id ..).const_mul (2 / π)).hasDerivWithinAt)
+    (fun x _ ↦ (hasDerivAt_cos ..).hasDerivWithinAt.sub_const _)
+    fun x hx ↦ neg_nonpos.2 $ sin_nonneg_of_mem_Icc $ Icc_subset_Icc_right (by linarith) $
+    interior_subset hx
+
+/-- **Jordan's inequality** for negative values. -/
+lemma sin_le_two_div_pi_mul (hx : -(π / 2) ≤ x) (hx₀ : x ≤ 0) : sin x ≤ 2 / π * x := by
+  simpa using two_div_pi_mul_le_sin (neg_nonneg.2 hx₀) (neg_le.2 hx)
+
+lemma cos_quadratic_upper_bound (hx : |x| ≤ π) : cos x ≤ 1 - 2 / π ^ 2 * x ^ 2 := by
+  wlog hx₀ : 0 ≤ x
+  · simpa using this (by rwa [abs_neg]) $ neg_nonneg.2 $ le_of_not_le hx₀
+  rw [abs_of_nonneg hx₀] at hx
+  -- TODO: `compute_deriv` tactic?
+  have hderiv (x) : HasDerivAt (fun x ↦ 1 - 2 / π ^ 2 * x ^ 2 - cos x) _ x :=
+    (((hasDerivAt_pow ..).const_mul _).const_sub _).sub $ hasDerivAt_cos _
+  simp only [Nat.cast_ofNat, Nat.succ_sub_succ_eq_sub, tsub_zero, pow_one, ← neg_sub', neg_sub,
+    ← mul_assoc] at hderiv
+  have hmono : MonotoneOn (fun x ↦ 1 - 2 / π ^ 2 * x ^ 2 - cos x) (Icc 0 (π / 2)) := by
+    refine monotoneOn_of_hasDerivWithinAt_nonneg (convex_Icc ..) (Continuous.continuousOn $
+      by continuity) (fun x _ ↦ (hderiv _).hasDerivWithinAt) fun x hx ↦ sub_nonneg.2 ?_
+    have ⟨hx₀, hx⟩ := interior_subset hx
+    calc 2 / π ^ 2 * 2 * x
+        = 2 / π * (2 / π * x) := by ring
```

Comment:
Is it standard to have the _ in front of the first = too?

Reply:
It is standard (and I prefer it that way) but Sébastien complained the other way around here: https://github.com/leanprover-community/mathlib4/pull/10525#discussion_r1489259915

Reply:
To clarify, I was suggesting leaving the initial expression and just adding _ to the second calc line.

Reply:
Can you write out explicitly what you mean?

Reply:
```lean
calc expr
  _ < … := by …
  _ > … := by …
```

But since that style thread isn’t resolved (and some people disagree with this) I’d say you have lots of flexibility. :)

Reply:
Yeah okay that's what I thought. Your syntax is (currently) invalid because it's equivalent to
```
calc
  expr _ < … := by …
  _ > … := by …
```
but `expr` isn't a function.

Reply:
That’s weird, because I use it a lot: https://github.com/girving/ray/blob/48d737369e856f925aa050a317d10ae95726804a/Ray/Misc/Bounds.lean#L142

Reply:
Uh, I am most surprised. I didn't know that was valid syntax